### ISSUE TITLE
🐞 bug(ethereum_clis): Incorrect exception handling

### ISSUE THREAD

**Issue by raxhvl** [ethereum/execution-spec-tests#971]
**BUG PRIORITY: LOW**

EEST has a module which maps errors from various EVM clients to a common set of exceptions. This mapping is not robust and this results in misleading exceptions being raised.

## Steps to reproduce

1. Create a new test file targeting Frontier fork.
```python
# file: tests/frontier/eip123_abc/test_abc.py
"""
A state test for testing exception.
"""

import pytest

from ethereum_test_tools import Alloc, Environment, StateTestFiller, Transaction

REFERENCE_SPEC_GIT_PATH = "DUMMY/eip-DUMMY.md"
REFERENCE_SPEC_VERSION = "DUMMY_VERSION"

@pytest.mark.valid_from("Frontier")
def test_abc(state_test: StateTestFiller, pre: Alloc):
    """
    Demo tx
    """
    env = Environment()

    tx = Transaction()

    # TODO: Add post-state allocations here.
    post: dict[str, dict] = {}

    state_test(env=env, pre=pre, post=post, tx=tx)
```

2. Fill the test
```sh
uv run fill tests/frontier/eip123_abc/test_abc.py
```
This will result in `TransactionException.TYPE_4_TX_CONTRACT_CREATION` exception which is
misleading since no contract was being created.

## Why the fill fails

In EEST, by default, a transaction is [protected](https://github.com/ethereum/execution-spec-tests/blob/c93a9bf24789a0a519cd37ea51c22c75fe474e26/src/ethereum_test_types/types.py#L699-L700) against replay. Support for [protected transactions](https://eips.ethereum.org/EIPS/eip-155) was added
AFTER Homestead (in [Spurious Dragon](https://eips.ethereum.org/EIPS/eip-607)), and so they get correctly rejected for Frontier and Homestead.

## Instance of incorrect exception mapping

The above unsupported transaction is rejected with the reason `Failed transaction:  `.

![Screenshot from 2024-11-22 12-16-31](https://github.com/user-attachments/assets/308238ee-2dc8-406b-bbe9-3f24fdb24a15)

This error is mapped by the following code:

https://github.com/ethereum/execution-spec-tests/blob/dac90491d0336728dec8b39260450f496d7f8447/src/ethereum_clis/clis/execution_specs.py#L133-L136

It appears that the code assumes failed transactions are only caused by failure to create a contract which is wrong, as shown here. 

Moreover, several other haphazard mappings expect partial error messages:

https://github.com/ethereum/execution-spec-tests/blob/dac90491d0336728dec8b39260450f496d7f8447/src/ethereum_clis/clis/execution_specs.py#L137-L152

We should have a robust error handling mechanism in EEST that produces reliable excpetions.

**Comment by felix314159** [comment]
I agree that our exception mapper is ugly but it is the most robust approach we have today. The logical order of checks exec clients perform can also vary which makes this problem even more difficult.
